xarray/backends/netCDF4_.py

```diff
@@ -485,20 +485,24 @@ def prepare_variable(
         if name in self.ds.variables:
             nc4_var = self.ds.variables[name]
         else:
-            nc4_var = self.ds.createVariable(
+            default_args = dict(
                 varname=name,
                 datatype=datatype,
                 dimensions=variable.dims,
-                zlib=encoding.get("zlib", False),
-                complevel=encoding.get("complevel", 4),
-                shuffle=encoding.get("shuffle", True),
-                fletcher32=encoding.get("fletcher32", False),
-                contiguous=encoding.get("contiguous", False),
-                chunksizes=encoding.get("chunksizes"),
+                zlib=False,
+                complevel=4,
+                shuffle=True,
+                fletcher32=False,
+                contiguous=False,
+                chunksizes=None,
                 endian="native",
-                least_significant_digit=encoding.get("least_significant_digit"),
+                least_significant_digit=None,
                 fill_value=fill_value,
             )
+            default_args.update(encoding)
```

Comment:
This will raise `AttributeError`s if the user supplies invalid encodings.
I personally think that is actually a better approach than the current one which silently ignores invalid encodings, but it might break some user codes. Maybe some other defs have preferences on this?

Reply:
The encoding keys are validated in the previous step with _extract_nc4_variable_encoding. If I get it right, this function will silently remove invalid keys.  It will raise a ValueError if raise_on_invalid is True, but it seems that it is always set to False. Maybe this will break the case when a compression key is used with an old version of netCF4 that does not support it, but this is probably not very common.

Reply:
> If I get it right, this function will silently remove invalid keys. I

Can you add a tesst for this please? or verify that a test exists

Reply:
I think this one already covers that behaviour https://github.com/pydata/xarray/blob/f63ede943840d1e51f16bd65c13c7957d7e84f6f/xarray/tests/test_backends.py#L4658